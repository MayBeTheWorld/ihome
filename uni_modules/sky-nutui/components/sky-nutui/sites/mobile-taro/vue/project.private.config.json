{
  "setting": {
    "compileHotReLoad": true
  },
  "description": "项目私有配置文件。此文件中的内容将覆盖 project.config.json 中的相同字段。项目的改动优先同步到此文件中。详见文档：https://developers.weixin.qq.com/miniprogram/dev/devtools/projectconfig.html",
  "condition": {
    "miniprogram": {
      "list": [
        {
          "name": "feedback/pages/notify/index",
          "pathName": "feedback/pages/notify/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "pages/index/index",
          "pathName": "pages/index/index",
          "query": "",
          "scene": null
        },
        {
          "name": "exhibition/pages/countdown/index",
          "pathName": "exhibition/pages/countdown/index",
          "query": "",
          "scene": null
        },
        {
          "name": "feedback/pages/collapse/index",
          "pathName": "feedback/pages/collapse/index",
          "query": "",
          "scene": null
        },
        {
          "name": "business/pages/signature/index",
          "pathName": "business/pages/signature/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "exhibition/pages/collapse/index",
          "pathName": "exhibition/pages/collapse/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        },
        {
          "name": "dentry/pages/textarea/index",
          "pathName": "dentry/pages/textarea/index",
          "query": "",
          "launchMode": "default",
          "scene": null
        }
      ]
    }
  },
  "projectname": "NutUI-Taro"
}