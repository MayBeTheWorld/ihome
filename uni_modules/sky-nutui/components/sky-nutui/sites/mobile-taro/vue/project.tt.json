{
  "miniprogramRoot": "./",
  "projectname": "@nutui/nutui-taro-mobile",
  "appid": "touristappid",
  "setting": {
    "es6": false,
    "minified": false
  }
}
