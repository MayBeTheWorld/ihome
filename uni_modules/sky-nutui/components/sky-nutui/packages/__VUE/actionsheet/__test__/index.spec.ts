import { config, mount } from '@vue/test-utils';
import ActionSheet from '../index.vue';
import NutIcon from '../../icon/index.vue';
import NutPopup from '../../popup/index.vue';
import { nextTick } from 'vue';

beforeAll(() => {
  config.global.components = {
    NutIcon,
    NutPopup
  };
});

afterAll(() => {
  config.global.components = {};
});

test('should render ActionSheet when visible is true', async () => {
  const wrapper = mount(ActionSheet, {
    props: {
      visible: true,
      isWrapTeleport: false,
      menuItems: [
        {
          name: '选项一'
        },
        {
          name: '选项二'
        },
        {
          name: '选项三'
        }
      ]
    }
  });
  const menuItem = wrapper.findAll('.nut-actionsheet-item');
  expect(menuItem.length).toBe(3);
});
test('should emit select event after clicking option', async () => {
  const wrapper = mount(ActionSheet, {
    props: {
      visible: true,
      isWrapTeleport: false,
      menuItems: [{ name: '选项一' }]
    }
  });
  const menuItem = wrapper.find('.nut-actionsheet-item');
  menuItem.trigger('click');
  await nextTick();
  expect(wrapper.emitted('choose')).toHaveLength(1);
  expect(wrapper.emitted('choose')![0]).toEqual([
    {
      name: '选项一'
    },
    0
  ]);
});

test('should render sure actionsheet when use custom props', () => {
  const wrapper = mount(ActionSheet, {
    props: {
      visible: true,
      isWrapTeleport: false,
      optionTag: 'names',
      optionSubTag: 'subname',
      menuItems: [{ names: '选项一', subname: '描述信息' }]
    }
  });

  expect(wrapper.html()).toMatchSnapshot();
  let subdesc = wrapper.find('.subdesc');
  expect(subdesc.html()).toContain('描述信息');
});

test('should render sure choose when use choose-tag-value', async () => {
  const wrapper = mount(ActionSheet, {
    props: {
      visible: true,
      isWrapTeleport: false,
      chooseTagValue: '选项一',
      menuItems: [{ name: '选项一' }, { name: '选项二' }]
    }
  });
  let item = wrapper.findAll<HTMLElement>('.nut-actionsheet-item');
  expect(item[0].element.style.color).toContain('238, 10, 36');
});

test('should render sure color when use color', async () => {
  const wrapper = mount(ActionSheet, {
    props: {
      visible: true,
      isWrapTeleport: false,
      chooseTagValue: '选项一',
      color: 'green',
      menuItems: [{ name: '选项一' }, { name: '选项二' }]
    }
  });
  let item = wrapper.findAll<HTMLElement>('.nut-actionsheet-item');
  expect(item[0].element.style.color).toContain('green');
});

test('should not emit select event after clicking disabled option', async () => {
  const wrapper = mount(ActionSheet, {
    props: {
      visible: true,
      isWrapTeleport: false,
      menuItems: [{ name: '选项一', disabled: true }]
    }
  });

  wrapper.find('.nut-actionsheet-item').trigger('click');
  await nextTick();
  expect(wrapper.emitted('select')).toBeFalsy();
});

test('should render description when use description', async () => {
  const wrapper = mount(ActionSheet, {
    props: {
      visible: true,
      isWrapTeleport: false,
      description: '这是一段描述信息',
      title: '主标题',
      menuItems: [{ name: '选项一', disabled: true }]
    }
  });
  let desc = wrapper.find('.desc');
  let title = wrapper.find('.nut-actionsheet-title');
  expect(desc.exists()).toBeTruthy();
  expect(title.exists()).toBeTruthy();
});

test('should emit cancel event after clicking cancel ', () => {
  const wrapper = mount(ActionSheet, {
    props: {
      visible: true,
      isWrapTeleport: false,
      menuItems: [{ name: '选项一' }],
      cancelTxt: '取消'
    }
  });
  wrapper.find('.nut-actionsheet-cancel').trigger('click');
  expect(wrapper.emitted('cancel')).toHaveLength(1);
});
