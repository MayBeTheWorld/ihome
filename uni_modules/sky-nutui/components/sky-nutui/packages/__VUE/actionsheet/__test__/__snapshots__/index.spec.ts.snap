// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render sure actionsheet when use custom props 1`] = `
"<view class=\\"nut-actionsheet\\">
  <view>
    <transition-stub class=\\"\\">
      <view class=\\"nut-overlay\\" style=\\"animation-duration: 0.3s; z-index: 2000;\\"></view>
    </transition-stub>
    <transition-stub>
      <view class=\\"nut-popup round popup-bottom popclass\\" style=\\"z-index: 2000; animation-duration: 0.3s;\\">
        <view class=\\"nut-actionsheet-panel\\">
          <!--v-if-->
          <view>
            <!--v-if-->
            <view class=\\"nut-actionsheet-menu\\">
              <view class=\\"nut-actionsheet-item\\">
                <view>选项一</view>
                <view class=\\"subdesc\\">描述信息</view>
              </view>
            </view>
            <!--v-if-->
          </view>
        </view>
        <!--v-if-->
      </view>
    </transition-stub>
  </view>
</view>"
`;
